doc/src/modules/physics/mechanics/symsystem.rst

```diff
@@ -0,0 +1,184 @@
+equations of motion formats. ::
+
+    >>> symsystem1 = system.SymbolicSystem(states, comb_explicit_rhs, 
+    ...                                    alg_con=alg_con_full, bodies=bodies, 
+    ...                                    loads=loads)
+    >>> symsystem2 = system.SymbolicSystem(states, comb_implicit_rhs, 
+    ...                                    mass_matrix=comb_implicit_mat,
+    ...                                    alg_con=alg_con_full,
+    ...                                    coord_idxs=(0, 1))
+    >>> symsystem3 = system.SymbolicSystem(states, dyn_implicit_rhs, 
+    ...                                    mass_matrix=dyn_implicit_mat,
+    ...                                    coordinate_derivatives=kin_explicit_rhs,
+    ...                                    alg_con=alg_con, coord_idxs=(0, 1), 
+    ...                                    speed_idxs=(2, 3))
+
+The `SymbolicSystem` class can determine which of the states are considered
+coordinates or speeds by passing in the indexes of the coordinates and speeds.
+If these indexes are not passed in the object will not be able to differentiate
+between coordinates and speeds. Like coordinates and speeds, the bodies and
+loads attributes can only be accessed if they are specified during
+initialization of the `SymbolicSystem` class. Lastly here are some attributes
+accessible from the `SymbolicSystem` class. ::
+
+    >>> symsystem1.states
+    Matrix([
+    [     x(t)],
+    [     y(t)],
+    [     u(t)],
+    [     v(t)],
+    [lambda(t)]])
+    >>> symsystem2.coordinates
+    Matrix([
+    [x(t)],
+    [y(t)]])
+    >>> symsystem3.speeds
+    Matrix([
+    [u(t)],
+    [v(t)]])
+    >>> symsystem1.comb_explicit_rhs
+    Matrix([
+    [                                   u(t)],
+    [                                   v(t)],
+    [(-g*y(t) + u(t)**2 + v(t)**2)*x(t)/l**2],
+    [(-g*y(t) + u(t)**2 + v(t)**2)*y(t)/l**2],
+    [   m*(-g*y(t) + u(t)**2 + v(t)**2)/l**2]])
+    >>> symsystem2.comb_implicit_rhs
+    Matrix([
+    [                       u(t)],
+    [                       v(t)],
+    [                          0],
+    [                          0],
+    [-g*y(t) + u(t)**2 + v(t)**2]])
+    >>> symsystem2.comb_implicit_mat
+    Matrix([
+    [1, 0, 0, 0,       0],
+    [0, 1, 0, 0,       0],
+    [0, 0, 1, 0, -x(t)/m],
+    [0, 0, 0, 1, -y(t)/m],
+    [0, 0, 0, 0,  l**2/m]])
+    >>> symsystem3.dyn_implicit_rhs
+    Matrix([
+    [                          0],
+    [                          0],
+    [-g*y(t) + u(t)**2 + v(t)**2]])
+    >>> symsystem3.dyn_implicit_mat
+    Matrix([
+    [1, 0, -x(t)/m],
+    [0, 1, -y(t)/m],
+    [0, 0,  l**2/m]])
+    >>> symsystem3.kin_explicit_rhs
+    Matrix([
+    [u(t)],
+    [v(t)]])
+    >>> symsystem1.alg_con
+    [4]
+    >>> symsystem1.dynamic_symbols
+    set([lambda(t), u(t), v(t), x(t), y(t)])
+    >>> symsystem1.constant_symbols
+    set([g, l, m])
```

Comment:
I think it was a bad choice for us to use sets in pydy.system.System for these. A tuple is nicer to work with because you can index it. A list would be fine too.